v2/cmux.go

```diff
@@ -73,3 +67,16 @@ type timeoutError struct{}
 func (e *timeoutError) Error() string   { return "i/o timeout" }
 func (e *timeoutError) Timeout() bool   { return true }
 func (e *timeoutError) Temporary() bool { return true }
+
+type Session interface {
+	OpenStream() (net.Conn, error)
+	AcceptStream() (net.Conn, error)
+	Close() error
+	NumStreams() int
+}
+
+type Protocol interface {
+	Client(net.Conn, *DialerOpts) (Session, error)
+	Server(net.Conn, *ListenOpts) (Session, error)
+	ErrorMapper() ErrorMapperFn
```

Comment:
Maybe I missed something, but how about just defining `TranslateErrors(error) error` in the interface? It would avoid introducing the concept of `ErrorMapperFn` and `ErrorMapper`.

Reply:
yeah good point